cmake_scripts/set_compiler_flags.cmake

```diff
@@ -0,0 +1,103 @@
+# Setting compiler secific flags
+INCLUDE(CheckCXXCompilerFlag)
+if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+
+	CHECK_CXX_COMPILER_FLAG(-ftemplate-depth=256 CLANG_TEMPLATE_DEPTH)
+	IF(CLANG_TEMPLATE_DEPTH)
+		MESSAGE(STATUS "Enabling '-ftemplate-depth=256' compiler flag required since boost 1.54.")
```

Comment:
We don't use boost anymore, is this stuff needed?  Presumably it's generic cmake boilerplate that makes useful things happen though